pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -910,17 +913,9 @@ impl RemoteTimelineClient {
             })
             .collect();
 
+        let not_referenced_count = remaining.len();
         if !remaining.is_empty() {
-            backoff::retry(
-                || async { self.storage_impl.delete_objects(&remaining).await },
-                |_e| false,
-                FAILED_UPLOAD_WARN_THRESHOLD,
-                FAILED_REMOTE_OP_RETRIES,
-                "delete_objects",
-                backoff::Cancel::new(shutdown_token(), || anyhow::anyhow!("Cancelled!")),
-            )
-            .await
-            .context("delete_objects")?;
+            self.deletion_queue_client.push_immediate(remaining).await?;
```

Comment:
> Under normal circumstances, including the planned seamless tenant migration work, not_referenced_count should be 0, right?

Right.  This is pre-existing code, and the existing "done deleting in timeline prefix" log message at the end of the function includes the count.

Refactoring this to remove the cleanup of unreferenced objects is orthogonal and we can pick it up elsewhere -- there is something kind of nice about having it here to provide a testable invariant that after deletion all the objects are gone, but the downside is that we end up having to flush the deletion queue in a place that we otherwise wouldn't.  That's for a future PR though.